route/measure.go

```diff
@@ -184,6 +184,14 @@ func Matrix(arcs [][]float64) ByIndex {
 	return matrixFunc(arcs)
 }
 
+// IsTriangular returns true if the triangle inequality holds for the provided
+// measure. It returns false if the measure does not implement the Triangular
+// interface or the triangle inequality does not hold.
+func IsTriangular(m any) bool {
```

Comment:
Do we really need this function when we have the Triangular interface? Why don't we require the type to be at least `func IsTriangular(m Triangular) bool`

Reply:
The reason is that the method checks if the measure `m` implements the interface, so if you know before the fact that a measure implements that interface you would not need this method. That is how I understand it at least.